### PR TITLE
fix handling of UUID columns containing values that are treated as a nan label

### PR DESCRIPTION
**Pull request by Anna050689**
fix handling of UUID columns containing values that are treated as a nan label

### CODE REVIEW

**Review by Ijka — COMMENTED** [review]
Shall we add to doc-string behaviour for nan-label?  
        
        If there are no NaNs and single non UUID/ULID value,
        it is treated as nan_label and set as NaN during training process

File: src/syngen/ml/vae/models/dataset.py [hunk 1]
```diff
@@ -665,62 +665,97 @@ def _is_valid_ulid(uuid):
...
+            non_uuid_values = [f"'{value}'" for value in non_uuid_values]
+            warning_msg += f", and non-UUID/ULID value/s - {', '.join(non_uuid_values)}"
+        if contain_nan:
+            warning_msg += ", and null value/s"
+        warning_msg += ". The column will be treated as a text column."
+        logger.warning(warning_msg)
+
+    def _handle_non_uuid_values(self, column, result, non_uuid_values):
+        """
+        Handle a column containing UUID/ULID columns
```
[the remark this answers is not in the document]
Comment (Anna050689) [reply src/syngen/ml/vae/models/dataset.py:729]: Thanks for the comment) I will fix it.

File: src/syngen/ml/vae/models/dataset.py [hunk 2]
```diff
@@ -665,62 +665,97 @@ def _is_valid_ulid(uuid):
...
-                    if uuid_obj.hex == i.replace("-", ""):
-                        return v
-                except ValueError:
-                    pass
-            return self._is_valid_ulid(i)
-
+    def _process_values(self, column: pd.DataFrame):
+        """
+        Process a text column and collect uuid types in the list - 'result',
+        and non uuid values into the list - 'non_uuid_values'
```
[the remark this answers is not in the document]
Comment (Anna050689) [reply src/syngen/ml/vae/models/dataset.py:684]: Thanks for the comment) I will fix it.

File: src/syngen/ml/vae/models/dataset.py [hunk 3]
```diff
@@ -665,62 +665,97 @@ def _is_valid_ulid(uuid):
...
 
-    def __handle_nan_label_in_uuid(self, x, non_uuid_values):
+    def _log_warning(self, column, non_uuid_values, contain_nan):
+        """
+        Build the log warning message related to columns containing UUID/ULID values
+        """
+        warning_msg = f"Column '{column.name}' contains UUID/ULID values"
+        if non_uuid_values:
+            non_uuid_values = [f"'{value}'" for value in non_uuid_values]
+            warning_msg += f", and non-UUID/ULID value/s - {', '.join(non_uuid_values)}"
```
[the remark this answers is not in the document]
Comment (Anna050689) [reply src/syngen/ml/vae/models/dataset.py:721]: I agree with you. I will fix it.

File: src/syngen/ml/vae/models/dataset.py [hunk 4]
```diff
@@ -669,24 +669,35 @@ def _is_valid_uuid(self, x):
         """
         Check if uuid_to_test is a valid UUID
```
[the remark this answers is not in the document]
Comment (Anna050689) [reply src/syngen/ml/vae/models/dataset.py]: Yes, I will add it.

**Review by Ijka — APPROVED** [review]

**Pull request merged**